diff -Naur util-linux-2.32/term-utils/agetty.c util-linux-2.32.tpg/term-utils/agetty.c
--- util-linux-2.32/term-utils/agetty.c	2018-03-06 11:35:18.991711000 +0000
+++ util-linux-2.32.tpg/term-utils/agetty.c	2018-03-29 06:39:21.843183474 +0000
@@ -1390,7 +1390,7 @@
 	 * Wait for a while, then read everything the modem has said so far and
 	 * try to extract the speed of the dial-in call.
 	 */
-	sleep(1);
+	usleep(1000);
 	if ((nread = read(STDIN_FILENO, buf, sizeof(buf) - 1)) > 0) {
 		buf[nread] = '\0';
 		for (bp = buf; bp < buf + nread; bp++)
@@ -1907,6 +1907,7 @@
 		/* Always show login prompt. */
 		write_all(STDOUT_FILENO, LOGIN, sizeof(LOGIN) - 1);
 	}
+	malloc_trim(0);
 }
 
 /* Select next baud rate. */
@@ -1952,7 +1953,7 @@
 	 * the baud rate).
 	 */
 	if ((op->flags & F_VCONSOLE) == 0)
-		sleep(1);
+		usleep(1000);
 	tcflush(STDIN_FILENO, TCIFLUSH);
 
 	eightbit = (op->flags & F_EIGHTBITS);
