diff -Naur util-linux-2.26/login-utils/chfn.c util-linux-2.26.tpg/login-utils/chfn.c
--- util-linux-2.26/login-utils/chfn.c	2015-02-10 09:49:46.000000000 +0000
+++ util-linux-2.26.tpg/login-utils/chfn.c	2015-02-21 00:40:12.950338813 +0000
@@ -143,7 +143,7 @@
 		{NULL, no_argument, 0, '0'},
 	};
 
-	while ((c = getopt_long(argc, argv, "f:r:p:h:o:uv", long_options,
+	while ((c = getopt_long(argc, argv, "f:r:w:p:h:o:uv", long_options,
 				&index)) != -1) {
 		switch (c) {
 		case 'f':
@@ -152,12 +152,14 @@
 			ctl->newf.full_name = optarg;
 			status += check_gecos_string(_("Name"), optarg);
 			break;
+		case 'r':
 		case 'o':
 			if (!ctl->allow_room)
 				errx(EXIT_FAILURE, _("login.defs forbids setting %s"), _("Office"));
 			ctl->newf.office = optarg;
 			status += check_gecos_string(_("Office"), optarg);
 			break;
+		case 'w':
 		case 'p':
 			if (!ctl->allow_work)
 				errx(EXIT_FAILURE, _("login.defs forbids setting %s"), _("Office Phone"));
