diff -Naur util-linux-2.25.2/lib/fileutils.c util-linux-2.25.2.tpg/lib/fileutils.c
--- util-linux-2.25.2/lib/fileutils.c	2014-09-16 12:37:06.000000000 +0000
+++ util-linux-2.25.2.tpg/lib/fileutils.c	2014-11-02 20:17:32.132751334 +0000
@@ -40,7 +40,7 @@
 		return -1;
 
 	old_mode = umask(077);
-	fd = mkostemp(localtmp, O_RDWR|O_CREAT|O_EXCL|O_CLOEXEC);
+	fd = mkstemp(localtmp);
 	umask(old_mode);
 	if (fd == -1) {
 		free(localtmp);
diff -Naur util-linux-2.25.2/libblkid/src/save.c util-linux-2.25.2.tpg/libblkid/src/save.c
--- util-linux-2.25.2/libblkid/src/save.c	2014-09-16 12:37:06.000000000 +0000
+++ util-linux-2.25.2.tpg/libblkid/src/save.c	2014-11-02 20:17:50.357246753 +0000
@@ -113,7 +113,7 @@
 		tmp = malloc(strlen(filename) + 8);
 		if (tmp) {
 			sprintf(tmp, "%s-XXXXXX", filename);
-			fd = mkostemp(tmp, O_RDWR|O_CREAT|O_EXCL|O_CLOEXEC);
+			fd = mkstemp(tmp);
 			if (fd >= 0) {
 				if (fchmod(fd, 0644) != 0)
 					DBG(SAVE, ul_debug("%s: fchmod failed", filename));
diff -Naur util-linux-2.25.2/libmount/src/utils.c util-linux-2.25.2.tpg/libmount/src/utils.c
--- util-linux-2.25.2/libmount/src/utils.c	2014-10-24 09:10:06.000000000 +0000
+++ util-linux-2.25.2.tpg/libmount/src/utils.c	2014-11-02 20:18:19.988052255 +0000
@@ -942,7 +942,7 @@
 	 */
 	oldmode = umask(S_IRGRP|S_IWGRP|S_IXGRP|
 			S_IROTH|S_IWOTH|S_IXOTH);
-	fd = mkostemp(n, O_RDWR|O_CREAT|O_EXCL|O_CLOEXEC);
+	fd = mkstemp(n);
 	if (fd < 0)
 		fd = -errno;
 	umask(oldmode);
