--- util-linux-2.30/sys-utils/hwclock-rtc.c.fix-ioctl~	2017-06-03 21:42:56.392274653 +0200
+++ util-linux-2.30/sys-utils/hwclock-rtc.c	2017-06-03 21:43:29.202504520 +0200
@@ -50,8 +50,8 @@ struct sparc_rtc_time
 	int month;	/* Month of year	1-12 */
 	int year;	/* Year			0-99 */
 };
-#define RTCGET _IOR('p', 20, struct sparc_rtc_time)
-#define RTCSET _IOW('p', 21, struct sparc_rtc_time)
+#define RTCGET _IOR('p', 20, sparc_rtc_time)
+#define RTCSET _IOW('p', 21, sparc_rtc_time)
 #endif
 
 /*
