--- util-linux-2.21/login-utils/chfn.c.chfnlsb	2012-01-09 15:53:10.000000000 +0100
+++ util-linux-2.21/login-utils/chfn.c	2012-02-26 03:04:50.450786636 +0100
@@ -221,7 +221,7 @@ static int parse_argv(int argc, char *ar
 	optind = 0;
 	info_given = false;
 	while (true) {
-		c = getopt_long(argc, argv, "f:r:p:h:o:uv", long_options,
+		c = getopt_long(argc, argv, "f:r:w:p:h:o:uv", long_options,
 				&index);
 		if (c == -1)
 			break;
@@ -245,10 +245,12 @@ static int parse_argv(int argc, char *ar
 			pinfo->full_name = optarg;
 			status = check_gecos_string(_("Name"), optarg);
 			break;
+		case 'r':
 		case 'o':
 			pinfo->office = optarg;
 			status = check_gecos_string(_("Office"), optarg);
 			break;
+		case 'w':
 		case 'p':
 			pinfo->office_phone = optarg;
 			status = check_gecos_string(_("Office Phone"), optarg);
