Index: disk-utils/fsck.cramfs.c
===================================================================
--- disk-utils/fsck.cramfs.c
+++ disk-utils/fsck.cramfs.c	2009-03-16 15:40:10.000000000 +0100
@@ -34,7 +34,7 @@
  */
 
 /* compile-time options */
-//#define INCLUDE_FS_TESTS	/* include cramfs checking and extraction */
+#define INCLUDE_FS_TESTS	/* include cramfs checking and extraction */
 
 #include <stdio.h>
 #include <stdarg.h>
