diff -up util-linux-2.36-rc1/login-utils/login.c.1~ util-linux-2.36-rc1/login-utils/login.c
--- util-linux-2.36-rc1/login-utils/login.c.1~	2020-06-28 02:19:57.835607135 +0200
+++ util-linux-2.36-rc1/login-utils/login.c	2020-06-28 02:20:25.627171973 +0200
@@ -585,7 +585,7 @@ static void log_lastlog(struct login_con
 	sa.sa_handler = SIG_IGN;
 	sigaction(SIGXFSZ, &sa, &oldsa_xfsz);
 
-	fd = open(_PATH_LASTLOG, O_RDWR, 0);
+	fd = open(_PATH_LASTLOG, O_RDWR | O_CREAT, 0);
 	if (fd < 0)
 		goto done;
 	offset = cxt->pwd->pw_uid * sizeof(ll);
