--- util-linux-2.30/login-utils/chfn.c.chfnlsb~	2017-05-19 11:43:34.780948977 +0200
+++ util-linux-2.30/login-utils/chfn.c	2017-06-03 21:42:33.733806522 +0200
@@ -147,7 +147,7 @@ static void parse_argv(struct chfn_contr
 		{ NULL, 0, NULL, 0 },
 	};
 
-	while ((c = getopt_long(argc, argv, "f:r:p:h:o:uv", long_options,
+	while ((c = getopt_long(argc, argv, "f:r:w:p:h:o:uv", long_options,
 				&index)) != -1) {
 		switch (c) {
 		case 'f':
@@ -156,12 +156,14 @@ static void parse_argv(struct chfn_contr
 			ctl->newf.full_name = optarg;
 			status += check_gecos_string(_("Name"), optarg);
 			break;
+		case 'r':
 		case 'o':
 			if (!ctl->allow_room)
 				errx(EXIT_FAILURE, _("login.defs forbids setting %s"), _("Office"));
 			ctl->newf.office = optarg;
 			status += check_gecos_string(_("Office"), optarg);
 			break;
+		case 'w':
 		case 'p':
 			if (!ctl->allow_work)
 				errx(EXIT_FAILURE, _("login.defs forbids setting %s"), _("Office Phone"));
