--- mount/swapon.c	2009-01-13 18:10:01.000000000 +0100
+++ mount/swapon.c.oden	2009-01-13 18:10:07.000000000 +0100
@@ -906,6 +906,8 @@ main_swapoff(int argc, char *argv[]) {
 					loop = opt + 5;
 				if (strncmp(opt, "encryption=", 11) == 0)
 					encryption = opt + 11;
+				if (strncmp(opt, "encrypted", 9) == 0)
+					skip=1;
 			}
 			if (loop && encryption) {
 				if (!is_in_proc_swaps(loop)) {  
