diff --git a/lib/fileutils.c b/lib/fileutils.c
index 92b474c..ff8bb86 100644
--- a/lib/fileutils.c
+++ b/lib/fileutils.c
@@ -37,7 +37,7 @@ int xmkstemp(char **tmpname, char *dir)
 		xasprintf(&localtmp, "%s/%s.XXXXXX", _PATH_TMP,
 			  program_invocation_short_name);
 	old_mode = umask(077);
-	fd = mkostemp(localtmp, O_RDWR|O_CREAT|O_EXCL|O_CLOEXEC);
+	fd = mkstemp(localtmp);
 	umask(old_mode);
 	if (fd == -1) {
 		free(localtmp);
diff --git a/libblkid/src/save.c b/libblkid/src/save.c
index 86eda6c..8f86e11 100644
--- a/libblkid/src/save.c
+++ b/libblkid/src/save.c
@@ -110,7 +110,7 @@ int blkid_flush_cache(blkid_cache cache)
 		tmp = malloc(strlen(filename) + 8);
 		if (tmp) {
 			sprintf(tmp, "%s-XXXXXX", filename);
-			fd = mkostemp(tmp, O_RDWR|O_CREAT|O_EXCL|O_CLOEXEC);
+			fd = mkstemp(tmp);
 			if (fd >= 0) {
 				if (fchmod(fd, 0644) != 0)
 					DBG(SAVE, blkid_debug("%s: fchmod failed", filename));
diff --git a/libmount/src/utils.c b/libmount/src/utils.c
index c328414..8d16420 100644
--- a/libmount/src/utils.c
+++ b/libmount/src/utils.c
@@ -841,7 +841,7 @@ int mnt_open_uniq_filename(const char *filename, char **name)
 	 */
 	oldmode = umask(S_IRGRP|S_IWGRP|S_IXGRP|
 			S_IROTH|S_IWOTH|S_IXOTH);
-	fd = mkostemp(n, O_RDWR|O_CREAT|O_EXCL|O_CLOEXEC);
+	fd = mkstemp(n);
 	umask(oldmode);
 
 	if (fd >= 0 && name)
